tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -981,20 +1007,28 @@ def do_test(self, test):
                                      self.protocol,
                                      url,
                                      test.timeout * self.timeout_multiplier,
-                                     self.extra_timeout).run()
+                                     self.extra_timeout,
+                                     test.extension).run()
 
         if success:
             data, extra = data
             return self.convert_result(test, data, extra=extra)
 
         return (test.make_result(*data), [])
 
-    def do_testharness(self, protocol, url, timeout):
+    def do_testharness(self, protocol, url, timeout, extension):
+        extension_id = None
+
         try:
             # The previous test may not have closed its old windows (if something
             # went wrong or if cleanup_after_test was False), so clean up here.
             protocol.testharness.close_old_windows()
+
+            if extension:
+                extension_id = protocol.web_extensions.load_web_extension(extension).get("extension")
+
             raw_results = self.run_testdriver(protocol, url, timeout)
```

Comment:
If you would follow an RFC process for the new test type, I think the loading of the extension might need to be in run_testdriver and not in do_testharness